### PR TITLE
Mostieri/export geometry utils interface

### PR DESCRIPTION
**Pull request by mariostieriansys**
Implementation of the geometry interface in the export utils

### CODE REVIEW

File: src/ansys/pyensight/core/utils/export.py [hunk 1]
```diff
@@ -464,3 +462,100 @@ def _animation_remote(
...
+        -------
+        bytes
+            Geometry export in bytes
+        """
+        rawdata = None
+        extension = self.extension_map.get(format)
+        if not extension:
+            raise RuntimeError("The geometry export format provided is not supported.")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            self._ensight.objs.core.PARTS.set_attr("SELECTED", True)
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/export.py:504]: This is the UI notion of selection, not command language (you are using savegeom() which is command language).  

self._ensight.part.select_all()

is more correct here.

File: src/ansys/pyensight/core/utils/export.py [hunk 2]
```diff
@@ -464,3 +462,100 @@ def _animation_remote(
...
+    def _geometry_remote(
+        self, format: str, begin_timestep: int = 0, end_timestep: int = 0, delta_timestep: int = 1
+    ) -> bytes:
+        """EnSight-side implementation.
+
+        Parameters
+        ----------
+        format : str
+            The format to export
+        begin_timestep: int
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/export.py:487]: General comment on the timesteps limits.  I kind of prefer the keywords:

starting_timestep
number_timesteps

(or 'frames').  The reason for this factorization is that for 95% of the case, I don't want users to have to figure out how many timesteps there are.  IMHO, the two common cases are:

1) all of the timesteps
2) the current timestep

With the current API, you need to look both of these up.  If you were to default the number_timesteps to 1 and default the starting_timestep to the current one, that handles case (2) w/o any keywords being set.  case (1) would only require that starting_timestep be set to 0 and the count be set to  ensight.objs.core.TIMESTEP_LIMITS[1].   An "nice" feature might be if number_timesteps is set to -1 or None explicitly, then we compute it to be from the start point to the last timestep.  Just trying to make sure the user can get away with setting as few of the keywords as possible. 

Another spin on this is that the default for the number of timesteps is None which is mapped to all of them, from the current timestep and the user can pretty easily set that to '1' for the curent timest, etc.

File: src/ansys/pyensight/core/utils/export.py [hunk 3]
```diff
@@ -464,3 +462,100 @@ def _animation_remote(
...
+        format: str = GEOM_EXPORT_GLTF,
+        begin_timestep: int = 0,
+        end_timestep: int = 0,
+        delta_timestep: int = 1,
+    ) -> None:
+        """Export a geometry file.
+
+        Parameters
+        ----------
+        filename: str
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/export.py:527]: If this is called from EnSight, it is not on the PyEnSight filesystem.  I would just drop that comment.

File: src/ansys/pyensight/core/utils/export.py (hunk 1 above)
Comment (mariostieriansys) [reply src/ansys/pyensight/core/utils/export.py:504]: uhm you're right, let me fix it

File: src/ansys/pyensight/core/utils/export.py (hunk 3 above)
Comment (mariostieriansys) [reply src/ansys/pyensight/core/utils/export.py:527]: yes, spurious comment, in a previous attempt I was trying to save it "locally" and then copy across

File: src/ansys/pyensight/core/utils/export.py (hunk 2 above)
Comment (mariostieriansys) [reply src/ansys/pyensight/core/utils/export.py:487]: So, I have changed the parameters like this:

        filename: str
            The location where to export the geometry
        format : str
            The format to export
        starting_timestep: int
            The first timestep to export. If None, defaults to the current timestep
        frames: int
            Number of timesteps to save. If None, defaults from the current timestep to the last
        delta_timestep: int
            The delta timestep to use when exporting

Comment (mariostieriansys) [reply src/ansys/pyensight/core/utils/export.py:487]: self,
        filename: str,
        format: str = GEOM_EXPORT_GLTF,
        starting_timestep: Optional[int] = None,
        frames: Optional[int] = 1,
        delta_timestep: Optional[int] = None

Comment (mariostieriansys) [reply src/ansys/pyensight/core/utils/export.py:487]: if not starting_timestep:
            starting_timestep = int(self._ensight.objs.core.TIMESTEP)
        if not frames or frames == -1:
            # Timesteps are 0-indexed so frames need to be increased of 1
            frames = int(self._ensight.objs.core.TIMESTEP_LIMITS[1]) + 1
        if not delta_timestep:
            delta_timestep = 1

**Pull request merged**